File: SDKs/googleapis.com/dlp/v2/python/src/sdk/models/shared/googleprivacydlpv2createstoredinfotyperequest.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import googleprivacydlpv2storedinfotypeconfig as shared_googleprivacydlpv2storedinfotypeconfig


@dataclass_json
@dataclasses.dataclass
class GooglePrivacyDlpV2CreateStoredInfoTypeRequest:
    r"""GooglePrivacyDlpV2CreateStoredInfoTypeRequest
    Request message for CreateStoredInfoType.
    """
    
    config: Optional[shared_googleprivacydlpv2storedinfotypeconfig.GooglePrivacyDlpV2StoredInfoTypeConfig] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('config') }})
    location_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('locationId') }})
    stored_info_type_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('storedInfoTypeId') }})
    

File: SDKs/googleapis.com/dlp/v2/java/lib/src/main/java/openapisdk/models/shared/GooglePrivacyDlpV2CreateStoredInfoTypeRequest.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * GooglePrivacyDlpV2CreateStoredInfoTypeRequest
 * Request message for CreateStoredInfoType.
**/
public class GooglePrivacyDlpV2CreateStoredInfoTypeRequest {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("config")
    public GooglePrivacyDlpV2StoredInfoTypeConfig config;
    public GooglePrivacyDlpV2CreateStoredInfoTypeRequest withConfig(GooglePrivacyDlpV2StoredInfoTypeConfig config) {
        this.config = config;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("locationId")
    public String locationId;
    public GooglePrivacyDlpV2CreateStoredInfoTypeRequest withLocationId(String locationId) {
        this.locationId = locationId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("storedInfoTypeId")
    public String storedInfoTypeId;
    public GooglePrivacyDlpV2CreateStoredInfoTypeRequest withStoredInfoTypeId(String storedInfoTypeId) {
        this.storedInfoTypeId = storedInfoTypeId;
        return this;
    }
}